cloud/services/domains.go

```diff
@@ -41,90 +45,53 @@ func AddIPToDNS(ctx context.Context, mscope *scope.MachineScope) error {
 	if err != nil {
 		return err
 	}
-	domainHostname := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier
 
-	// Create/Update the A record for this IP and name combo
-	for ipType, publicIP := range publicIPs {
-		if err := CreateUpdateDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
+	for _, dnsEntry := range dnsEntries {
+		if operation == "delete" {
+			if err := DeleteDomainRecord(ctx, mscope, domainID, dnsEntry); err != nil {
+				return err
+			}
+			continue
+		}
+		if err := CreateUpdateDomainRecord(ctx, mscope, domainID, dnsEntry); err != nil {
 			return err
 		}
 	}
 
-	// Create/Update the TXT record for this IP and name combo
-	txtRecordValueString := createSHA256HashOfString(mscope.LinodeMachine.Name)
-	if err := CreateUpdateDomainRecord(ctx, mscope, domainHostname, "owner:"+txtRecordValueString, dnsTTLSec, domainID, "TXT"); err != nil {
-		return err
-	}
-
 	return nil
 }
 
-// DeleteIPFromDNS deletes the A and TXT record for the machine
-func DeleteIPFromDNS(ctx context.Context, mscope *scope.MachineScope) error {
+// getDNSEntriesToEnsure return DNS entries to create/delete
+func (d *DNSEntries) getDNSEntriesToEnsure(mscope *scope.MachineScope) ([]DNSOptions, error) {
+	d.mux.Lock()
+	defer d.mux.Unlock()
 	dnsTTLSec := rutil.DefaultDNSTTLSec
 	if mscope.LinodeCluster.Spec.Network.DNSTTLSec != 0 {
 		dnsTTLSec = mscope.LinodeCluster.Spec.Network.DNSTTLSec
 	}
 
-	// Check if instance is a control plane node
-	if !kutil.IsControlPlaneMachine(mscope.Machine) {
-		return nil
-	}
-
-	// Get the public IP that was assigned
-	publicIPs, err := getMachinePublicIPs(ctx, mscope)
-	if err != nil {
-		return err
-	}
-
-	// Get domainID from domain name
-	domainID, err := GetDomainID(ctx, mscope)
-	if err != nil {
-		return err
+	if mscope.LinodeMachine.Status.Addresses == nil {
+		return nil, fmt.Errorf("no addresses available on the LinodeMachine resource")
 	}
 	domainHostname := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier
 
-	// Delete A record
-	for ipType, publicIP := range publicIPs {
-		if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
-			return err
+	for _, IPs := range mscope.LinodeMachine.Status.Addresses {
+		recordType := "A"
```

Comment:
nit: we should probably use https://pkg.go.dev/github.com/linode/linodego#DomainRecordType, e.g.
```suggestion
		recordType := linodego.RecordTypeA
```